app/views/ssh_keys/_form.html.erb

```diff
@@ -0,0 +1,18 @@
+<%= form_for @ssh_key, :url => user_ssh_keys_path(:user_id => @user) do |f| %>
+  <%= base_errors_for @ssh_key %>
+  <% if @ssh_key.errors.include?(:fingerprint) || @ssh_key.errors.include?(:length) %>
+    <p><%= _('Please correct the error(s) below and submit your changes again.') %></p>
+
+    <div class="alert alert-warning alert-block base">
+      <% [:fingerprint, :length].each do |field|%>
+        <% @ssh_key.errors.full_messages_for(field).each do |e| %>
+          <li><%= e %></li>
+        <% end %>
+      <% end %>
+    </div>
+  <% end %>
+  <%= textarea_f f, :key, :size => "col-md-8" , :rows=> '3' %>
```

Comment:
@ohadlevy : I think, the upload idea is good. The ACE editor doesn't bring much value imho.
Anyways, both features aren't really necessary to get the core functionality in. @dLobatog already suggested to split the API part in a separate PR. I think, we should keep this out of this PR and create follow-up tickets.